x-pack/plugins/endpoint/server/routes/metadata/metadata.test.ts

```diff
@@ -37,15 +41,33 @@ describe('test endpoint route', () => {
       IClusterClient
     >;
     mockScopedClient = elasticsearchServiceMock.createScopedClusterClient();
+    mockSavedObjectClient = savedObjectsClientMock.create();
     mockClusterClient.asScoped.mockReturnValue(mockScopedClient);
     routerMock = httpServiceMock.createRouter();
     mockResponse = httpServerMock.createResponseFactory();
     registerEndpointRoutes(routerMock, {
+      indexPatternRetriever: createMockMetadataIndexPatternRetriever(),
       logFactory: loggingServiceMock.create(),
       config: () => Promise.resolve(EndpointConfigSchema.validate({})),
     });
   });
 
+  function createRouteHandlerContext(
+    dataClient: jest.Mocked<IScopedClusterClient>,
+    savedObjectsClient: jest.Mocked<SavedObjectsClientContract>
+  ) {
+    return ({
+      core: {
+        elasticsearch: {
+          dataClient,
+        },
+        savedObjects: {
+          client: savedObjectsClient,
+        },
+      },
+    } as unknown) as RequestHandlerContext;
```

Comment:
whats the reason for the unsafe cast here? if its required, can you add a comment explaining it.

Reply:
I think `unknown` is needed here because the object that is defined doesn't fully qualify as a `RequestHandlerContext` it's missing some fields. I wonder if we can fix this using a mock instead 🤔